[timeseries] Improve error message when specifying an invalid path to load

As of v1.1, the error message produced by TimeSeriesPredictor when loaded with an invalid path is confusing. We should make sure that users understand that the `predictor.pkl` was unable to found, as that is more important than not finding the `version.txt` file.

## TimeSeries

### Code

```
from autogluon.timeseries import TimeSeriesPredictor
TimeSeriesPredictor.load("some_invalid_path/")
```

### Exception

```
WARNING: Could not find version file at "some_invalid_path/version.txt".
This means that the predictor was fit in an AutoGluon version `<=0.7.0`.

############################## WARNING ##############################
WARNING: AutoGluon version differs from the version used to create the predictor! This may lead to instability and it is highly recommended the predictor be loaded with the exact AutoGluon version it was created with. AutoGluon does not support backwards compatibility.
	Predictor Version: Unknown (Likely <=0.7.0)
	Current Version:   1.1.1b20240601
############################## WARNING ##############################

Traceback (most recent call last):
  File "/home/ubuntu/workspace/code/scratch/scratch_run.py", line 77, in <module>
    predictor = TimeSeriesPredictor.load("some_invalid_path/")
  File "/home/ubuntu/workspace/code/autogluon/timeseries/src/autogluon/timeseries/predictor.py", line 1110, in load
    check_saved_predictor_version(
  File "/home/ubuntu/workspace/code/autogluon/common/src/autogluon/common/utils/utils.py", line 162, in check_saved_predictor_version
    raise AssertionError(
AssertionError: Predictor was created on version Unknown (Likely <=0.7.0) but is being loaded with version 1.1.1. Please ensure the versions match to avoid instability. While it is NOT recommended, this error can be bypassed by specifying `require_version_match=False`. Exceptions encountered after setting `require_version_match=False` may be very cryptic, and in most cases mean that the predictor is fully incompatible with the installed version.

```

## Tabular

### Code
```
from autogluon.tabular import TabularPredictor
TabularPredictor.load("some_invalid_path/")
```

### Exception
```
WARNING: Could not find version file at "some_invalid_path/version.txt".
This means that the predictor was fit in an AutoGluon version `<=0.3.1`.
Traceback (most recent call last):
  File "/home/ubuntu/workspace/code/scratch/scratch_run.py", line 79, in <module>
    predictor = TabularPredictor.load("some_invalid_path/")
  File "/home/ubuntu/workspace/code/autogluon/tabular/src/autogluon/tabular/predictor/predictor.py", line 4440, in load
    predictor = cls._load(path=path)
  File "/home/ubuntu/workspace/code/autogluon/tabular/src/autogluon/tabular/predictor/predictor.py", line 4363, in _load
    predictor: TabularPredictor = load_pkl.load(path=os.path.join(path, cls.predictor_file_name))
  File "/home/ubuntu/workspace/code/autogluon/common/src/autogluon/common/loaders/load_pkl.py", line 43, in load
    with compression_fn_map[compression_fn]["open"](validated_path, "rb", **compression_fn_kwargs) as fin:
FileNotFoundError: [Errno 2] No such file or directory: 'some_invalid_path/predictor.pkl'
```

--- a/timeseries/src/autogluon/timeseries/predictor.py
+++ b/timeseries/src/autogluon/timeseries/predictor.py
@@ -1098,6 +1098,10 @@ class TimeSeriesPredictor(TimeSeriesPredictorDeprecatedMixin):
             raise ValueError("`path` cannot be None or empty in load().")
         path: str = setup_outputdir(path, warn_if_exist=False)
 
+        predictor_path = Path(path) / cls.predictor_file_name
+        if not predictor_path.exists():
+            raise FileNotFoundError(f"No such file '{predictor_path}'")
+
         try:
             version_saved = cls._load_version_file(path=path)
         except:
@@ -1116,7 +1120,7 @@ class TimeSeriesPredictor(TimeSeriesPredictorDeprecatedMixin):
 
         logger.info(f"Loading predictor from path {path}")
         learner = AbstractLearner.load(path)
-        predictor = load_pkl.load(path=os.path.join(learner.path, cls.predictor_file_name))
+        predictor = load_pkl.load(path=str(predictor_path))
         predictor._learner = learner
         predictor.path = learner.path
         return predictor

--- a/timeseries/tests/unittests/test_predictor.py
+++ b/timeseries/tests/unittests/test_predictor.py
@@ -1751,3 +1751,7 @@ def test_when_predictor_saved_to_same_directory_and_loaded_then_number_of_models
         == set(loaded_predictor.model_names())
         == set(hyperparameters).union({"WeightedEnsemble"})
     )
+
+def test_when_invalid_path_provided_to_load_then_correct_exception_is_raised():
+    with pytest.raises(FileNotFoundError, match="No such file"):
+        TimeSeriesPredictor.load("some_invalid_path")

--- a/timeseries/tests/unittests/test_predictor.py
+++ b/timeseries/tests/unittests/test_predictor.py
@@ -1752,6 +1752,7 @@ def test_when_predictor_saved_to_same_directory_and_loaded_then_number_of_models
         == set(hyperparameters).union({"WeightedEnsemble"})
     )
 
+
 def test_when_invalid_path_provided_to_load_then_correct_exception_is_raised():
     with pytest.raises(FileNotFoundError, match="No such file"):
         TimeSeriesPredictor.load("some_invalid_path")

